addons/payment_worldline/models/payment_provider.py

```diff
@@ -114,10 +121,16 @@ def _worldline_calculate_signature(self, method, endpoint, content_type, dt_rfc)
         :param str endpoint: The endpoint to be reached by the request.
         :param str content_type: The 'Content-Type' header of the request.
         :param datetime.datetime dt_rfc: The timestamp of the request, in RFC1123 format.
+        :param str idempotency_key: The idempotency key to pass in the request.
         :return: The calculated signature.
         :rtype: str
         """
-        values_to_sign = [method, content_type, dt_rfc, f'/v2/{self.worldline_pspid}/{endpoint}']
+        # specific order required: method, content_type, date, custom headers, endpoint
+        values_to_sign = [method, content_type, dt_rfc]
+        if idempotency_key:
```

Comment:
If `idempotency_key` can be omitted from the method call, it should be a keyword argument in the signature and passed with `idempotency_key=idempotency_key` in the call.